tools/azsdk-cli/docs/specs/0-scenario-2.spec.md

```diff
@@ -0,0 +1,818 @@
+
+Test assets created for 5/5 languages.
+```
+
+### 7. Set Test Mode
+
+**Command:**
+
+```bash
+azsdk test-mode set --mode live-record
+```
+
+**Options:**
+
+- `--mode <live|live-record|playback>`: Testing mode to set (required)
+- `--package-path <path>`: Path to specific SDK package (default: current directory)
+- `--languages <list>`: Space-separated list of languages (default: detected from package)
+
+**Expected Output:**
+
+```text
+Setting test mode to live-record...
+
+✓ Test mode set to live-record
+
+Ready to run tests with recording enabled.
+```
+
+### 8. Run Tests
+
+**Command:**
+
+```bash
+azsdk package run-tests --package-path <path_to_sdk_package>/
+```
+
+**Options:**
+
+- `--package-path <path>`: Path to the specific SDK package directory (required)
+- `--test-filter <pattern>`: Run only tests matching pattern
+- `--parallel`: Run tests in parallel (when supported)
+- `--verbose`: Show detailed test output
+
+**Expected Output:**
+
+```text
+Running tests for Health Deidentification SDK...
+
+✓ Python tests passed (12 recordings captured)
+✓ .NET tests passed (10 recordings captured)
+
+Test execution complete. Recordings saved to language-specific directories.
+```
+
+### 9. Teardown Test Resources
+
+**Command:**
+
+```bash
+azsdk test-resources manage --service healthdataaiservices --action delete
+```
+
+**Options:**
+
+- `--service <name>`: Service short name (required)
+- `--action <create|delete>`: Action to perform (required)
+- `--resource-group <name>`: Override the default resource group naming convention
+- `--subscription <id>`: Target a specific Azure subscription
+
+**Expected Output:**
+
+```text
+Tearing down test resources for healthdataaiservices...
+
+✓ Resource group azsdk-healthdataaiservices-tests deleted
+
+Test resources cleaned up successfully.
+```
+
+---
+
+## Pipeline & CI Usage
+
+Scenario 2 also needs to consider **how the CLI tools will be used inside CI/CD pipelines** (for example, GitHub Actions, Azure Pipelines), even though the exact requirements are still evolving.
+
+### Goals
+
+- Enable **non-interactive**, repeatable runs of the Scenario 2 workflow using CLI commands.
+- Make it easy to plug `azsdk` commands into existing SDK build/test pipelines with minimal glue.
+- Ensure environment setup and customization flows behave predictably in headless environments.
+
+### Early Requirements & Considerations
+
+- `azsdk_package_setup_env` and `azsdk_verify_setup`:
+  - Must support **exit codes** that cleanly distinguish "all good", "fixable issues", and "blocking failures".
+  - Should be configurable via flags or environment variables in CI (for example, `--no-install` or `AZSDK_SETUP_ENV_MODE=verify-only`) to avoid surprise installations on shared agents.
+- Customization and generation:
+  - `azsdk_package_customize_typespec`, `azsdk_package_generate_code`, and `azsdk_package_customized_code_update` should be chainable in a single job with clear, machine-readable logs.
```

Comment:
If we want to integrate customization part into the pipeline, it means we will have subsequent steps to create separate PRs against both the spec repo and SDK repo to include the changes made by the customization tools.